src/main/java/com/rarchives/ripme/ripper/rippers/InstagramRipper.java

```diff
@@ -125,9 +125,15 @@ private String getOriginalUrl(String imageURL) {
     private String getMedia(JSONObject data) {
         String imageURL = "";
         if (data.has("videos")) {
-            imageURL = data.getJSONObject("videos").getJSONObject("standard_resolution").getString("url");
+        	JSONObject videosObject = data.getJSONObject("videos");
+			if (!videosObject.isNull("standard_resolution")) {
+        		imageURL = videosObject.getJSONObject("standard_resolution").getString("url");
+        	}
         } else if (data.has("images")) {
-           imageURL = data.getJSONObject("images").getJSONObject("standard_resolution").getString("url");
+        	JSONObject imagesObject = data.getJSONObject("images");
+			if (!imagesObject.isNull("standard_resolution")) {
+        		imageURL = imagesObject.getJSONObject("standard_resolution").getString("url");
+        	}
```

Comment:
Is it really necessary to make an object of different names (`videosObject`, `imagesObject`) for this ? This is just pure code duplication to me.

Reply:
To me it looks clearer this way. If I extract the object before the if I can reduce the number of words by one. If some other fields are needed based on the type of the object this becomes useful.